src/script/sign.cpp

```diff
@@ -614,7 +614,10 @@ bool IsSegWitOutput(const SigningProvider& provider, const CScript& script)
 {
     std::vector<valtype> solutions;
     auto whichtype = Solver(script, solutions);
-    if (whichtype == TxoutType::WITNESS_V0_SCRIPTHASH || whichtype == TxoutType::WITNESS_V0_KEYHASH || whichtype == TxoutType::WITNESS_UNKNOWN) return true;
+    if (whichtype == TxoutType::WITNESS_V0_SCRIPTHASH ||
```

Comment:
nit: Maybe turn this into a switch-case to get compiler output when v2 is added, but this isn't updated?

Reply:
@sipa Are you still working on this? Happy to take over if not.